### PR TITLE
FIX-#5273: Make `ParquetFileToRead` a named tuple

### LINKED ISSUE #5273: unidist on MPI fails to serialize a class derived from `NamedTuple`
Originally found in https://github.com/modin-project/modin/pull/5059#discussion_r983666941. If we revert `ParquetFileToRead` back to be derived from `NamedTuple`, tests for parquet fail on unidist on MPI. We should investigate and fix the issue.

The failed job is https://github.com/modin-project/modin/actions/runs/3524150780/jobs/5909761234.

### PR DESCRIPTION
**Pull request by Retribution98**
Signed-off-by: Kirill Suvorov <kirill.suvorov@intel.com>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5273 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: requirements/env_unidist.yml [hunk 1]
```diff
@@ -51,4 +51,4 @@ dependencies:
       - flake8-no-implicit-concat
       - flake8-print
       # The `numpydoc` version should match the version installed in the `lint-pydocstyle` job of the CI.
-      - numpydoc==1.1.0
+      - numpydoc==1.1.0
```
Comment (anmyachev) [inline requirements/env_unidist.yml:54]: This should help revert the change:
```suggestion
      - numpydoc==1.1.0

```

**Review by YarShev — APPROVED** [review]
